### ISSUE TITLE
Allow user to establish his preferred locale

### ISSUE THREAD

**Issue by bertocq** [consul/consul#2435]
At the user account (`/account`) list all available languages in a dropdown, defaulting to the current default locale, so the user can set his preferred one.

Override current I18n.locale with current_user.preferred_locale value.

For existing languages use `config.i18n.available_locales` and for default locale `config.i18n.default_locale`. Use config.i18n.default_locale to set the default value for all existing users `preferred_locale` attributes on the migration.

**Comment by bertocq** [comment]
As I was re-reading & researching the codebase to make this issue easier to be taken by @juandefelix ... I realized we already have this feature 😱  (almost):

The topbar language selector lets the user choose among all the app available languages... and if the user is logged in (not an anonymous visitor) his `locale` attribute will be set/updated with the selected language (check [app/controllers/application_controller.rb#L57-L59](https://github.com/consul/consul/blob/master/app/controllers/application_controller.rb#L57-L59)).
![screen shot 2018-02-08 at 22 57 40](https://user-images.githubusercontent.com/983242/36000525-7f7cac24-0d23-11e8-82bc-61ef02eda79f.jpg)

It was my fault to just copy & pasted a dictated requirement (two times) without actually researching if this feature already existed (it was introduced by https://github.com/consul/consul/pull/851).

Now I think that the request came from a similar point of view... it may not be clear to the user that changing the language on that dropdown will make the app remember his selection for future visits. But I'm not sure duplicating that dropdown as well under `/account` will achieve it... neither I can remember from who/where we got that request 🤔 I'm summoning @decabeza here for both UI/UX advice on the matter (and maybe some remembering as well)

🙏 sorry @juandefelix!
